cluster/node.go

```diff
@@ -75,18 +117,150 @@ func (n *Node) SetState(s NodeState) {
 	n.Unlock()
 }
 
-// provide thread safe json Marshaling
-func (n *Node) MarshalJSON() ([]byte, error) {
+func (n *Node) SetPartitions(part []int32) {
```

Comment:
Yes, currently SetPartitions() is only every called once.  But the point is that it is only called once due to limitations in the input plugins.  There is no reason to enforce the partitions to only be set once.